mmv1/third_party/terraform/tests/resource_network_connectivity_service_connection_policies_test.go

```diff
@@ -0,0 +1,134 @@
+package google
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+)
+
+func TestAccNetworkConnectivityServiceConnectionPolicy_update(t *testing.T) {
+	t.Parallel()
+
+	networkName := fmt.Sprintf("tf-test-network-%s", RandString(t, 10))
+	networkProducerName := fmt.Sprintf("tf-test-network-%s", RandString(t, 10))
+	subnetworkConsumerName := fmt.Sprintf("tf-test-subnet-consumer-%s", RandString(t, 10))
+	subnetworkProducerName := fmt.Sprintf("tf-test-subnet-producer-%s", RandString(t, 10))
+	serviceConnectionPolicyName := fmt.Sprintf("tf-test-service-connection-policy-%s", RandString(t, 10))
+
+	VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckNetworkConnectivityServiceConnectionPolicyDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccNetworkConnectivityServiceConnectionPolicy_basic(networkName, subnetworkConsumerName, networkProducerName, subnetworkProducerName, serviceConnectionPolicyName),
+			},
+			{
+				ResourceName:      "google_network_connectivity_service_connection_policy.default",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccNetworkConnectivityServiceConnectionPolicy_update(networkName, subnetworkConsumerName, networkProducerName, subnetworkProducerName, serviceConnectionPolicyName),
+			},
+			{
+				ResourceName:      "google_network_connectivity_service_connection_policy.default",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccNetworkConnectivityServiceConnectionPolicy_basic(networkName, subnetworkConsumerName, networkProducerName, subnetworkProducerName, serviceConnectionPolicyName),
+			},
+			{
+				ResourceName:      "google_network_connectivity_service_connection_policy.default",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccNetworkConnectivityServiceConnectionPolicy_basic(networkName, subnetworkConsumerName, networkProducerName, subnetworkProducerName, serviceConnectionPolicyName string) string {
```

Comment:
Nit: If you have the time and feel like it, we have a pattern that works a little better when there are more arguments, where you use a `context` variable and `NPrintf()`. It mainly helps prevent mistakes if the positional arguments are passed in the wrong order. However, it's definitely ok to leave it as-is.